test/Microsoft.AspNet.Mvc.FunctionalTests/TestHelper.cs

```diff
@@ -58,9 +59,8 @@ public static TestServer CreateServer(Action<IApplicationBuilder> builder, strin
         {
             applicationPath = applicationPath ?? WebsitesDirectoryPath;
 
-            // Get current IApplicationEnvironment; likely added by the host.
             var provider = services.BuildServiceProvider();
-            var originalEnvironment = provider.GetRequiredService<IApplicationEnvironment>();
+            var originalEnvironment = PlatformServices.Default.Application;
```

Comment:
We get `originalEnvironment` to wrap it in our `TestApplicationEnvironment`. Will the `TestApplicationEnvironment` instance added just below be picked up correctly?